engine/pkg/externalresource/resourcetypes/interfaces.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcetypes
+
+import (
+	"context"
+
+	resModel "github.com/pingcap/tiflow/engine/pkg/externalresource/resourcemeta/model"
+)
+
+// ResourceTypeController is an interface providing relevant operations related
+// to one resource type.
+type ResourceTypeController interface {
```

Comment:
IMHO, why not the `ResourceTypeHandler` since it only has `GetHandler` interface?

Reply:
More methods are planned in future PRs.